readme.md
[npm-badge]: https://img.shields.io/npm/v/react-native-indicators.svg?colorB=ff6d00
[npm-url]: https://npmjs.com/package/react-native-indicators
[license-badge]: https://img.shields.io/npm/l/react-native-indicators.svg?colorB=448aff
[license-url]: https://raw.githubusercontent.com/n4kz/react-native-indicators/master/license.txt
[travis-badge]: https://api.travis-ci.org/n4kz/react-native-indicators.svg?branch=master
[travis-url]: https://travis-ci.org/n4kz/react-native-indicators?branch=master
[example-url]: https://user-images.githubusercontent.com/2055622/28246049-e82c70e8-6a1b-11e7-93cc-8aa6d0d19867.gif
[indicator]: https://facebook.github.io/react-native/docs/activityindicator.html
[v014-url]: https://github.com/n4kz/react-native-indicators/releases/tag/0.14.0

# react-native-indicators

[![npm][npm-badge]][npm-url]
[![license][license-badge]][license-url]
[![travis][travis-badge]][travis-url]

Activity indicator collection for React Native

![example][example-url]

## Features

* Easy to use
* Consistent look and feel on iOS and Android
* Can be used as drop-in replacement for [ActivityIndicator][indicator]
* Configurable appearance and animation
* Configurable indicator size
* Native driver based animation
* Pure javascript implementation
* Compatible with RN 0.45+, for older versions use [0.14.0][v014-url]

## Installation

```bash
npm install --save react-native-indicators
```

## Usage

```javascript
import React, { Component } from 'react';
import {
  BallIndicator,
  BarIndicator,
  DotIndicator,
  MaterialIndicator,
  PacmanIndicator,
  PulseIndicator,
  SkypeIndicator,
  UIActivityIndicator,
  WaveIndicator,
} from 'react-native-indicators';

class Example extends Component {
  render() {
    return (
      <DotIndicator color='white' />
    );
  }
}
```

## Common properties

 name               | description                   | type     | default
:------------------ |:----------------------------- | --------:|:-------------
 animationEasing    | Animation easing function     | Function | Easing.linear
 animationDuration  | Animation duration in ms      |   Number | 1200
 animating          | Animation toggle              |  Boolean | true
 interaction        | Animation is interaction      |  Boolean | true
 hidesWhenStopped   | Hide when not animating       |  Boolean | true

## BallIndicator properties

 name  | description         | type   | default
:----- |:------------------- | ------:|:------------
 color | Component color     | String | rgb(0, 0, 0)
 count | Component count     | Number | 8
 size  | Base component size | Number | 40

## BarIndicator properties

 name  | description         | type   | default
:----- |:------------------- | ------:|:------------
 color | Component color     | String | rgb(0, 0, 0)
 count | Component count     | Number | 3
 size  | Base component size | Number | 40

## DotIndicator properties

 name  | description         | type   | default
:----- |:------------------- | ------:|:------------
 color | Component color     | String | rgb(0, 0, 0)
 count | Component count     | Number | 4
 size  | Base component size | Number | 16

## MaterialIndicator properties

 name       | description           | type   | default
:---------- |:--------------------- | ------:|:------------
 color      | Component color       | String | rgb(0, 0, 0)
 size       | Base component size   | Number | 40
 trackWidth | Indicator track width | Number | size / 10

## PacmanIndicator properties

 name  | description         | type   | default
:----- |:------------------- | ------:|:------------
 color | Component color     | String | rgb(0, 0, 0)
 size  | Base component size | Number | 48

## PulseIndicator properties

 name  | description         | type   | default
:----- |:------------------- | ------:|:------------
 color | Component color     | String | rgb(0, 0, 0)
 size  | Base component size | Number | 40

## SkypeIndicator properties

 name     | description             | type   | default
:-------- |:----------------------- | ------:|:------------
 color    | Component color         | String | rgb(0, 0, 0)
 count    | Component count         | Number | 5
 size     | Base component size     | Number | 40
 minScale | Minimum component scale | Number | 0.2
 maxScale | Maximum component scale | Number | 1.0

## UIActivityIndicator properties

 name  | description         | type   | default
:----- |:------------------- | ------:|:------------
 color | Component color     | String | rgb(0, 0, 0)
 count | Component count     | Number | 12
 size  | Base component size | Number | 40

## WaveIndicator properties

 name       | description         | type   | default
:---------- |:------------------- | ------:|:------------
 color      | Component color     | String | rgb(0, 0, 0)
 count      | Component count     | Number | 4
 size       | Base component size | Number | 40
 waveFactor | Wave base number    | Number | 0.54
 waveMode   | Wave appearance     | String | fill

Possible values for `waveMode` are `fill` and `outline`

## Example

```bash
git clone https://github.com/n4kz/react-native-indicators
cd react-native-indicators/example
npm install
npm run ios # or npm run android
```

## Copyright and License

BSD License

Copyright 2017-2019 Alexander Nazarov. All rights reserved.


changelog.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [0.17.0] - 2019-10-16

### Changed

- PacmanIndicator animation

## [0.16.0] - 2019-10-01

### Added

- `hidesWhenStopped` prop

### Changed

- MaterialIndicator animation

## [0.15.0] - 2019-09-28

### Added

- `trackWidth` prop for MaterialIndicator
- Animation state restoration for Indicator component

## [0.14.0] - 2019-09-26

### Fixed

- WaveIndicator glitch on Android
- Deprecation warnings for React component lifecycle methods

[Unreleased]: https://github.com/n4kz/react-native-indicators/compare/0.17.0...HEAD
[0.17.0]: https://github.com/n4kz/react-native-indicators/compare/0.16.0...0.17.0
[0.16.0]: https://github.com/n4kz/react-native-indicators/compare/0.15.0...0.16.0
[0.15.0]: https://github.com/n4kz/react-native-indicators/compare/0.14.0...0.15.0
[0.14.0]: https://github.com/n4kz/react-native-indicators/compare/0.13.0...0.14.0


index.d.ts
declare module 'react-native-indicators' {
  import { Animated, EasingFunction } from 'react-native';
  import { Component } from 'react';
  interface BaseProps {
    /**
     * Animation easing function
     * @default Easing.linear
     */
    animationEasing?: EasingFunction;

    /**
     * Animation duration in ms
     * @default1200
     */
    animationDuration?: number;
  }

  export interface UIActivityIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;
    /**
     * Component count
     * @default 12
     */
    count?: number;
    /**
     * Base component size
     * @default 40
     */
    size?: number;
  }

  export class UIActivityIndicator extends Component<
    UIActivityIndicatorProps
    > { }

  export interface BallIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;
    /**
     * Component count
     * @default 8
     */
    count?: number;
    /**
     * Base component size
     * @default 40
     */
    size?: number;
  }

  export class BallIndicator extends Component<BallIndicatorProps> { }

  export interface BarIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;
    /**
     * Component count
     * @default 3
     */
    count?: number;
    /**
     * Base component size
     * @default 40
     */
    size?: number;
  }

  export class BarIndicator extends Component<BarIndicatorProps> { }

  export interface DotIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;
    /**
     * Component count
     * @default 4
     */
    count?: number;
    /**
     * Base component size
     * @default 16
     */
    size?: number;
  }

  export class DotIndicator extends Component<DotIndicatorProps> { }

  export interface DotIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;
    /**
     * Component count
     * @default 4
     */
    count?: number;
    /**
     * Base component size
     * @default 16
     */
    size?: number;
  }

  export class DotIndicator extends Component<DotIndicatorProps> { }

  export interface MaterialIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;

    /**
     * Base component size
     * @default 40
     */
    size?: number;
  }

  export class MaterialIndicator extends Component<MaterialIndicatorProps> { }

  export interface PulseIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;

    /**
     * Base component size
     * @default 40
     */
    size?: number;
  }

  export class PulseIndicator extends Component<PulseIndicatorProps> { }

  export interface PacmanIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;

    /**
     * Base component size
     * @default 48
     */
    size?: number;
  }

  export class PacmanIndicator extends Component<PacmanIndicatorProps> { }

  export interface PulseIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;

    /**
     * Base component size
     * @default 40
     */
    size?: number;
  }

  export class PulseIndicator extends Component<PulseIndicatorProps> { }

  export interface SkypeIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;
    /**
     * Component count
     * @default 5
     */
    count?: number;
    /**
     * Base component size
     * @default 40
     */
    size?: number;
    /**
     * Minimum component scale
     * @default 0.2
     */
    minScale?: number;
    /**
     * Maximum component scale
     * @default 1.0
     */
    maxScale?: number;
  }

  export class SkypeIndicator extends Component<SkypeIndicatorProps> { }

  export interface WaveIndicatorProps extends BaseProps {
    /**
     * Component color
     * @default 'rgb(0, 0, 0)'
     */
    color?: string;
    /**
     * Component count
     * @default 4
     */
    count?: number;
    /**
     * Base component size
     * @default 40
     */
    size?: number;
    /**
     * Minimum component scale
     * @default 0.54
     */
    waveFactor?: number;
    /**
     * Maximum component scale
     * @default 'fill'
     */
    waveMode?: 'fill' | 'outline';
  }

  export class WaveIndicator extends Component<WaveIndicatorProps> { }
}
